Fluid/Scope.cs

```diff
@@ -66,41 +66,67 @@ public FluidValue GetValue(string name)
                 : NilValue.Instance;
         }
 
+        /// <summary>
+        /// Deletes the value with the specified name in the chain of scopes.
+        /// </summary>
+        /// <param name="name">The name of the value to delete.</param>
         public void Delete(string name)
+        {
+            if (!_forLoopScope)
+            {
+                DeleteOwn(name);
+            }
+            else
+            {
+                Parent.Delete(name);
+            }
+        }
+
+        /// <summary>
+        /// Deletes the value with the specified name in the current scopes.
+        /// </summary>
+        /// <param name="name">The name of the value to delete.</param>
+        public void DeleteOwn(string name)
         {
             if (_properties != null)
             {
-                if (!_forLoopScope)
-                {
-                    _properties.Remove(name);
-                }
-                else
-                {
-                    Parent.Delete(name);
-                }
+                _properties.Remove(name);
             }
         }
 
+        /// <summary>
+        /// Sets the value with the specified name in the chain of scopes.
+        /// </summary>
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public void SetValue(string name, FluidValue value)
         {
             if (!_forLoopScope)
             {
-                _properties ??= new Dictionary<string, FluidValue>();
-
-                _properties[name] = value ?? NilValue.Instance;
+                SetOwnValue(name, value);
             }
             else
             {
                 Parent.SetValue(name, value);
             }
         }
 
+        /// <summary>
+        /// Sets the value with the specified name in the current scope.
+        /// </summary>
+        public void SetOwnValue(string name, FluidValue value)
```

Comment:
maybe ` [MethodImpl(MethodImplOptions.AggressiveInlining)]`?

Reply:
What's your approach with that, I am tending to let the JIT decide these days because I don't have a benchmark that proves it better. But do you think it can never be worse than not having it?

Reply:
JIT can help and be wise, but probably not on full framework. I made the comment mostly because you converted from direct field access to method so call site would more likely have old behavior with aggressive inlining.

Reply:
Good point. I assume it doesn't hurt to add it whatsoever so I am adding it back.